### PR TITLE
bucket tutorial

### PR DESCRIPTION
**Pull request by kmdeck**
# PULL REQUEST

## Purpose and Content
Add tutorial for the bucket model in standalone and coupled mode.

## Benefits and Risks
Explains the bucket model and shows how to run it.

### CODE REVIEW

**Comment by kmdeck** [comment]
Generated docs here:
https://clima.github.io/ClimaLSM.jl/previews/PR81/generated/bucket_tutorial/
https://clima.github.io/ClimaLSM.jl/previews/PR81/generated/coupled_bucket/

File: docs/tutorials/bucket_tutorial.jl [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+# ``
+
+# ``
+# F_{bot} = -κ \frac{T_{sfc} - T_{base}}{d}
+# ``
+
+# where `I` is the infiltration as defined in [1], `P_liq` (m/s) is the
+# water volume flux of precipitation, `E` (m/s) is the water volume flux
+# in evaporation, `ρc` is the volumetric
+# heat capacity of the land,  `R_n` is the net radiation, `SHF` the sensibl
```
Comment (LenkaNovak) [inline docs/tutorials/bucket_tutorial.jl:49]: sensible

File: docs/tutorials/bucket_tutorial.jl [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+# required across models (e.g. the density of water and ice, the latent
+# heat of fusion at a reference temperature, etc). The land model requires
+# additional parameters as described in the text above. These two sets
+# are combined in the object `BucketModelParameters` as follows:
+import ClimaLSM
+include(joinpath(pkgdir(ClimaLSM), "parameters", "create_parameters.jl"));
+earth_param_set = create_lsm_parameters(FT);
+
+# Define our `BulkAlbedo` model using a constant soil and snow albedo:
+# The soil albedo is a function of your coordinates, which would be
```
Comment (LenkaNovak) [inline docs/tutorials/bucket_tutorial.jl:128]: your -> our ?

File: docs/tutorials/bucket_tutorial.jl [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    S_c,
+    W_f,
+    z_0m,
+    z_0b,
+    earth_param_set,
+);
+
+# Set up the model domain. At every coordinate point, we'll solve
+# an ODE for `W`, `Ws`, and `T_sfc`. In coupled simulations run at the same
+# resolution as the atmosphere, the bucket domain would match the
```
Comment (LenkaNovak) [inline docs/tutorials/bucket_tutorial.jl:163]: I don't know if you'd like to include it here, but technically the coupled run should be able to deal with different resolutions too.

File: docs/tutorials/bucket_tutorial.jl [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+saved_values = SavedValues(FT, ClimaCore.Fields.FieldVector);
+cb = SavingCallback(
+    (u, t, integrator) -> copy(integrator.p),
+    saved_values;
+    saveat = 0:Δt:tf,
+);
+
+sol = solve(prob, Euler(); dt = Δt, saveat = 0:Δt:tf, callback = cb);
+
+# Extractin the solution from what is returned by the ODE.jl commands
```
Comment (LenkaNovak) [inline docs/tutorials/bucket_tutorial.jl:266]: extracting

File: docs/tutorials/bucket_tutorial.jl [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+# is a bit clunky right now, but we are working on hiding some of this.
+# `parent` extracts the underlying data from the ClimaCore.Field.FieldVector,
```
Comment (LenkaNovak) [inline docs/tutorials/bucket_tutorial.jl:268]: ClimaCore.Field*s*. Also, would `parent()` of a `FieldVector` work? I've only ever applied it to individual `Field` objects.

File: docs/tutorials/coupled_bucket.jl [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+# please see [the bucket model tutorial](https://clima.github.io/ClimaLSM.jl/dev/generated/bucket_tutorial/).
+
+# This tutorial shows how to set up a simulation for a coupled simulation. More detail for coupled runs can be
+# found in the ClimaCoupler.jl [documentation](https://clima.github.io/ClimaCoupler.jl/dev/). In
+# preparation for understanding this tutorial, we recommend also reading the [intro to multi-component models tutorial](https://clima.github.io/ClimaLSM.jl/dev/generated/LSM_single_column_tutorial/) as well as being familiar
+# with multiple dispatch programming in Julia.
+
+# # Background
+
+# Recall that in order to drive the system in standalone mode,
```
Comment (LenkaNovak) [inline docs/tutorials/coupled_bucket.jl:13]: I wonder about removing the discussion about setting up the standalone bucket in this tutorial, or at least shortening it a bit.

File: docs/tutorials/coupled_bucket.jl [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+        R_n = p.bucket.R_n,
+        LHF = p.bucket.LHF,
+        SHF = p.bucket.SHF,
+        E = p.bucket.E,
+    )
+end
+
+# Essentially, this method simply returns the values stored in the auxiliary state `p`. Importantly, this function is
+# called by the bucket model 
+# each time step **after** the coupler has already computed these values and modifed `p`!
```
Comment (LenkaNovak) [inline docs/tutorials/coupled_bucket.jl:127]: computed (or extracted from another model)

**Review by LenkaNovak — APPROVED** [review]
Looks good, thanks @kmdeck ! I just left a few minor comments after which it's good to go.

**Comment by kmdeck** [comment]
> Looks good, thanks @kmdeck ! I just left a few minor comments after which it's good to go.

thank you!!

File: docs/tutorials/bucket_tutorial.jl (hunk 5 above)
Comment (kmdeck) [reply docs/tutorials/bucket_tutorial.jl]: I dont think that it would! but ive never tried either. I changed the language to be clearer.

**Pull request merged**